src/main/java/org/entur/lamassu/metrics/MetricsService.java

```diff
@@ -130,4 +100,63 @@ public void registerEntityCount(String entity, int entityCount) {
       logger.warn("entity unknown entity={}", entity);
     }
   }
+
+  private AtomicInteger getSubscriptionFailedSetupCounter(FeedProvider feedProvider) {
+    return getCounter(
+      feedProvider,
+      subscriptionFailedSetupCounters,
+      SUBSCRIPTION_FAILEDSETUP
+    );
+  }
+
+  private AtomicInteger getMissingRequiredFilesCounter(FeedProvider feedProvider) {
+    return getCounter(
+      feedProvider,
+      validationFeedErrorsCounters,
+      VALIDATION_MISSING_REQUIRED_FILES
+    );
+  }
+
+  private AtomicInteger getValidationFeedErrorsCounter(FeedProvider feedProvider) {
+    return getCounter(
+      feedProvider,
+      validationMissingRequiredFilesCounters,
+      VALIDATION_FEED_ERRORS
+    );
+  }
+
+  private AtomicInteger getCounter(
+    FeedProvider feedProvider,
+    Map<String, AtomicInteger> counterMap,
+    String metricName
+  ) {
+    AtomicInteger counter;
+    if (counterMap.containsKey(feedProvider.getSystemId())) {
```

Comment:
~~I think we require either a Map of Maps of use systemId+metricName as key (preferred)~~ Sorry, the counterMap is already per metricName